internal/app/tfsec/block/hclattribute.go

```diff
@@ -474,20 +474,56 @@ func (attr *HCLAttribute) Reference() (*Reference, error) {
 	case *hclsyntax.RelativeTraversalExpr:
 		switch s := t.Source.(type) {
 		case *hclsyntax.IndexExpr:
-			collectionRef := createDotReferenceFromTraversal(s.Collection.Variables()...)
+			collectionRef, err := createDotReferenceFromTraversal(s.Collection.Variables()...)
+			if err != nil {
+				return nil, err
+			}
 			key, _ := s.Key.Value(attr.ctx)
 			collectionRef.SetKey(key)
 			return collectionRef, nil
 		default:
-			return createDotReferenceFromTraversal(t.Source.Variables()...), nil
+			return createDotReferenceFromTraversal(t.Source.Variables()...)
 		}
 	case *hclsyntax.ScopeTraversalExpr:
-		return createDotReferenceFromTraversal(t.Traversal), nil
+		return createDotReferenceFromTraversal(t.Traversal)
+	case *hclsyntax.TemplateExpr:
+		refs := attr.ReferencesInTemplate()
+		if len(refs) == 0 {
+			return nil, fmt.Errorf("no references in template")
+		}
+		return refs[0], nil
 	default:
 		return nil, fmt.Errorf("not a reference: no scope traversal")
 	}
 }
 
+func (attr *HCLAttribute) AllReferences() []*Reference {
+	refs := attr.ReferencesInTemplate()
+	if len(refs) > 0 {
+		return refs
+	}
+	ref, err := attr.Reference()
+	if err != nil {
+		return nil
+	}
+	return append(refs, ref)
+}
+
+func (attr *HCLAttribute) ReferencesInTemplate() []*Reference {
```

Comment:
does it need to be public and if yes, should it be on the interface?

Reply:
good point, it's private now